script.js

```diff
@@ -0,0 +1,166 @@
+var cartLen = 0;
+var cost = 0;
+
+async function setup()
+{
+	var jsonLink= "https://raw.githubusercontent.com/alexsimkovich/patronage/main/api/data.json";
+	let textObj = await fetch(jsonLink);
+	let txt = await textObj.text();
+	var obj = JSON.parse(txt);
+
+	document.getElementById("subBtn").style.display='none';
+
+	makePizzasListItems(obj.length);
+	var str=[];
+	for(var i = 0; i<obj.length; i++)
+	{
+			setImage(obj[i].image,i+1);
+			setName(obj[i].title,i+1);
+			setPrice(obj[i].price,i+1);
+			setIngredients(obj[i].ingredients,i+1);
+			setButton(i+1);
+	}
+}
```

Comment:
Kod jest bardzo skomplikowany, można to łatwo usprawnić. W HTML-u masz `<ul id="pizzas_list"></ul>`, więc możesz odrazu wyrenderować elementy, iterując po tablice z danymi:

```
const dataContainer = document.getElemetById('pizzas_list')
response.forEach(item => {
  dataContainer.insertAdjacentHTML('afterbegin', `
    <img class="image" src='${item.image}' />
    <div class="name">${item.name}</div>
    <div class="price">${item.price} zł</div>
    <div class="ingredients">${item.ingredients.join(', ')}</div>
`);
})
```

Wtedy funkcje setImage, setName, etc. nie są potrzebne